File: leetcode/js/intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  const result = new Set();
  const nums2Set = new Set(nums2);

  for (const num of nums1) {
    if (nums2Set.has(num)) {
      result.add(num);
    }
  }

  return [...result];
};

File: leetcode/rust/intersection-of-two-arrays.rs
impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        use std::collections::HashSet;

        let nums1: HashSet<i32> = nums1.into_iter().collect();
        let nums2: HashSet<i32> = nums2.into_iter().collect();

        let result = nums1.intersection(&nums2);

        result.cloned().collect::<Vec<i32>>()
    }
}
